pkg/controller/controller.go

```diff
@@ -277,13 +303,32 @@ func (p *csiProvisioner) Provision(options controller.VolumeOptions) (*v1.Persis
 		req.ControllerCreateSecrets = credentials
 		secret.Name, secret.Namespace, _ = getSecretAndNamespaceFromParameters(options.Parameters)
 	}
-	ctx, cancel := context.WithTimeout(context.Background(), p.timeout)
-	defer cancel()
 
-	rep, err := p.csiClient.CreateVolume(ctx, &req)
+	opts := wait.Backoff{Duration: backoffDuration, Factor: backoffFactor, Steps: backoffSteps}
+	err = wait.ExponentialBackoff(opts, func() (bool, error) {
+		ctx, cancel := context.WithTimeout(context.Background(), p.timeout)
+		defer cancel()
+		rep, err = p.csiClient.CreateVolume(ctx, &req)
```

Comment:
This is ok, but still depends on the csiCLient to be setup. That is the #64 where this client should always be setup and never stop retrying. It looks like this change does not handler #64. If it does not handle that issue and the client is never setup correctly, then `.CreateVolume` will never work. and will eventually time out. 

It looks like `Provision()` and maybe other calls needs to create the connection to the client at every time it calls.

Reply:
Good point, up to @sbezverk if he wants to address it in this PR or in a seperate PR.

Reply:
@lpabon I am sorry but I do disagree with csi client `constantly` trying to connect. Here is why: We do not carry state of relation between provisioner and the driver. So every time when there is a need for provisioner to communicate, then it will build connection to the provisioner, check it is `actual` state of mind and run the operation.  I have done lots of negative testing with this approach and it worked really well. If you can reproduce a failure scenario which is not coverer here, please let me know I would be really interested to dissect it  ;)

Reply:
Ok, I no problem. Let's go with this change and we can adjust the `Connect()` call if we need to.